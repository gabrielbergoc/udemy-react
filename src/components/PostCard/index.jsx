import P from 'prop-types';
import './styles.css';

const PostCard = ({ post }) => (
  <div className="post">
    <img src={post.cover} alt={post.title}></img>
    <div className="post-content">
      <h2>{post.title}</h2>
      <p>{post.body}</p>
    </div>
  </div>
);

PostCard.propTypes = {
  post: P.shape({
    id: P.number.isRequired,
    title: P.string.isRequired,
    body: P.string.isRequired,
    cover: P.string.isRequired,
  }).isRequired,
};

export default PostCard;
