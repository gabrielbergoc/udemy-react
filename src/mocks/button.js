export const buttonProps = {
  text: 'button text',
  title: 'button title',
  onClick: jest.fn(),
  disabled: false,
};
