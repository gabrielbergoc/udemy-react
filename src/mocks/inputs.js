export const searchInputProps = {
  value: "a value",
  handleChange: jest.fn(),
  placeholder: "a placeholder",
}
