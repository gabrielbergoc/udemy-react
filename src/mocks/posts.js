export const rawPosts = [
  {
    userId: 1,
    id: 1,
    title: 'post #1 title',
    body: 'post #1 body',
  },
  {
    userId: 1,
    id: 2,
    title: 'post #2 title',
    body: 'post #2 body',
  },
  {
    userId: 1,
    id: 3,
    title: 'post #3 title',
    body: 'post #3 body',
  },
  {
    userId: 1,
    id: 4,
    title: 'post #4 title',
    body: 'post #4 body',
  },
  {
    userId: 1,
    id: 5,
    title: 'post #5 title',
    body: 'post #5 body',
  },
  {
    userId: 1,
    id: 6,
    title: 'post #6 title',
    body: 'post #6 body',
  },
  {
    userId: 1,
    id: 7,
    title: 'post #7 title',
    body: 'post #7 body',
  },
  {
    userId: 1,
    id: 8,
    title: 'post #8 title',
    body: 'post #8 body',
  },
  {
    userId: 1,
    id: 9,
    title: 'post #9 title',
    body: 'post #9 body',
  },
  {
    userId: 1,
    id: 10,
    title: 'post #10 title',
    body: 'post #10 body',
  },
];

export const photos = [
  {
    albumId: 1,
    id: 1,
    title: 'photo #1 title',
    url: 'img/img1.png',
    thumbnailUrl: 'https://via.placeholder.com/150/92c952',
  },
  {
    albumId: 1,
    id: 2,
    title: 'photo #2 title',
    url: 'img/img2.png',
    thumbnailUrl: 'https://via.placeholder.com/150/771796',
  },
  {
    albumId: 1,
    id: 3,
    title: 'photo #3 title',
    url: 'img/img3.png',
    thumbnailUrl: 'https://via.placeholder.com/150/24f355',
  },
  {
    albumId: 1,
    id: 4,
    title: 'photo #4 title',
    url: 'img/img4.png',
    thumbnailUrl: 'https://via.placeholder.com/150/d32776',
  },
  {
    albumId: 1,
    id: 5,
    title: 'photo #5 title',
    url: 'img/img5.png',
    thumbnailUrl: 'https://via.placeholder.com/150/f66b97',
  },
  {
    albumId: 1,
    id: 6,
    title: 'photo #6 title',
    url: 'img/img6.png',
    thumbnailUrl: 'https://via.placeholder.com/150/56a8c2',
  },
  {
    albumId: 1,
    id: 7,
    title: 'photo #7 title',
    url: 'img/img7.png',
    thumbnailUrl: 'https://via.placeholder.com/150/b0f7cc',
  },
  {
    albumId: 1,
    id: 8,
    title: 'photo #8 title',
    url: 'img/img8.png',
    thumbnailUrl: 'https://via.placeholder.com/150/54176f',
  },
  {
    albumId: 1,
    id: 9,
    title: 'photo #9 title',
    url: 'img/img9.png',
    thumbnailUrl: 'https://via.placeholder.com/150/51aa97',
  },
  {
    albumId: 1,
    id: 10,
    title: 'photo #10 title',
    url: 'img/img10.png',
    thumbnailUrl: 'https://via.placeholder.com/150/810b14',
  },
];

export const posts = rawPosts.map((post, i) => {
  const postWithCover = { ...post };
  postWithCover.cover = photos[i].url;
  return postWithCover;
});

export const post = posts[0];
