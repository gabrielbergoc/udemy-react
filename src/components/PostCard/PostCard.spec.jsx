import { render, screen } from '@testing-library/react';
import PostCard from '.';
import { post } from '../../mocks/posts';

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard post={post} />);

    const img = screen.getByRole('img', { name: /title/i });
    const heading = screen.getByRole('heading', { name: /title/i });
    const p = screen.getByText(/body/i);

    expect(img).toHaveAttribute('src', 'img/img1.png');
    expect(heading).toBeInTheDocument();
    expect(p).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { container } = render(<PostCard post={post} />);

    expect(container).toMatchSnapshot();
  });
});
