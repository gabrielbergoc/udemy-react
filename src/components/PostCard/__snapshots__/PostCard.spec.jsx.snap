// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match the snapshot 1`] = `
<div>
  <div
    class="post"
  >
    <img
      alt="post #1 title"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        post #1 title
      </h2>
      <p>
        post #1 body
      </p>
    </div>
  </div>
</div>
`;
